goStr_test.go

```diff
@@ -280,3 +280,60 @@ func TestBeforeLast(t *testing.T) {
 		}
 	}
 }
+
+func TestCurrency(t *testing.T) {
+	decimalTrue := true
+	decimalFalse := false
+	dotSeparatorTrue := true
+	dotSeparatorFalse := false
+	spacerTrue := true
+	spacerFalse := false
+
+	testCases := []struct {
+		amount       interface{}
+		code         CurrencyCode
+		useDecimal   *bool
+		dotSeparator *bool
+		useSpacer    *bool
+		expected     string
+	}{
+		{10000, CURRENCY_IDR, &decimalTrue, &dotSeparatorFalse, &spacerTrue, "Rp 10,000.00"},
+		{"10000", CURRENCY_USD, &decimalTrue, &dotSeparatorFalse, &spacerTrue, "$ 10,000.00"},
+		{12000, CURRENCY_IDR, &decimalTrue, &dotSeparatorTrue, &spacerTrue, "Rp 12.000,00"},
+		{"12000", CURRENCY_IDR, &decimalFalse, &dotSeparatorFalse, &spacerTrue, "Rp 12,000"},
+		{15250.8972, CURRENCY_IDR, &decimalTrue, &dotSeparatorTrue, &spacerFalse, "Rp15.250,90"},
```

Comment:
currency should be strict, never round a decimal value. if the input 15250.8972, the output should be Rp15.250,89

Reply:
> currency should be strict, never round a decimal value. if the input 15250.8972, the output should be Rp15.250,89

resoved and tested 👍